Grad-Projects/my-planner
# Server Setup

**This is the Backend** 💥 with localhost

## Test
Sure, you can use postman or insomnia, etc but

I would prefer you test the API in vscode:  download  the extension:  **REST Client extension by Huachao Mao**
There is a **test.rest** file in the **Testing** folder that shows all endpoints that have been created. 

### Conditions

1. Ensure you have all the packages necessary: cd into backend and just run **npm install**
2. Ensure you have a database matching the details in **database_connection.js**
3. There is a script in the Testing Folder, **Flyway_Combined_DBsetup.sql** that combines all the flyway queries, you can use it to setup your local database with the test data

### How to test

- cd into backend and run npm start in the terminal to start the server
- Go to the Testing Folder and open the test.rest file.
- You should see a clickable "send request" text above each endpoint example.
There is _at least one_ endpoint example for each endpoint created.

## Todo


1. **HIGH PRIORITY**:  finish the getUserEmail() function in cognito_utils.js by  authenticating the user and getting their email from cognito.
2.  **HIGH PRIORITY**:  Make the API work with official base url and aws rds database 
3.   Any other endpoints needed.
4.   Any security enhancements
